temporian/core/data/sampling.py

```diff
@@ -53,3 +68,14 @@ def is_unix_timestamp(self) -> bool:
     @creator.setter
     def creator(self, creator: Optional[Operator]):
         self._creator = creator
+
+    def _parse_index(
+        self, index: List[Union[Tuple[str, IndexDtypes], IndexLevel]]
+    ) -> List[IndexLevel]:
+        if all([isinstance(index_level, IndexLevel) for index_level in index]):
+            return index
+
+        return [
+            IndexLevel(index_name, index_dtype)
+            for index_name, index_dtype in index
+        ]
```

Comment:
very nit: we could iterate over the list just once and for each item check if its an `IndexLevel`, else convert it, with a single comprehension